clients/python/llmengine/file.py

```diff
@@ -0,0 +1,192 @@
+    def get(cls, file_id: str) -> GetFileResponse:
+        """
+        Get filename and size of a file.
+
+        Args:
+            file_id (`str`):
+                ID of the file
+
+        Returns:
+            GetFileResponse: an object that contains the ID, filename, and size of the requested file
+
+        === "Getting metadata about file in Python"
+            ```python
+            from llmengine import File
+
+            response = File.get(
+                file_id="file-abc123",
+            )
+
+            print(response.json())
+            ```
+
+        === "Response in JSON"
+            ```json
+            {
+                "id": "file-abc123",
+                "filename": "training_dataset.csv",
+                "size": 100
+            }
+            ```
+        """
+        response = cls._get(f"v1/files/{file_id}", timeout=DEFAULT_TIMEOUT)
+        return GetFileResponse.parse_obj(response)
+
+    @classmethod
+    def list(cls) -> ListFilesResponse:
+        """
+        List all files, with information about their filenames and sizes.
+
+        Returns:
+            ListFilesResponse: an object that contains a list of all files and their filenames and sizes
+
+        === "Listing files in Python"
+            ```python
+            from llmengine import File
+
+            response = File.list()
+            print(response.json())
+            ```
+
+        === "Response in JSON"
+            ```json
+            {
+                "files": [
+                    {
+                        "id": "file-abc123",
+                        "filename": "training_dataset.csv",
+                        "size": 100
+                    },
+                    {
+                        "id": "file-def456",
+                        "filename": "validation_dataset.csv",
+                        "size": 50
+                    }
+                ]
+            }
+            ```
+        """
+        response = cls._get("v1/files", timeout=30)
+        return ListFilesResponse.parse_obj(response)
+
+    @classmethod
+    def delete(cls, file_id: str) -> DeleteFileResponse:
+        """
+        Deletes a file.
+
+        Args:
+            file_id (`str`):
+                ID of the file
+
+        Returns:
+            DeleteFileResponse: an object that contains whether the deletion was successful
+
+        === "Deleting file in Python"
+            ```python
+            from llmengine import File
+
+            response = File.delete(file_id="file-abc123")
+            print(response.json())
+            ```
+
+        === "Response in JSON"
+            ```json
+            {
+                "deleted": true
+            }
+            ```
+        """
+        response = cls._delete(
+            f"v1/files/{file_id}",
+            timeout=DEFAULT_TIMEOUT,
+        )
+        return DeleteFileResponse.parse_obj(response)
+
+    @classmethod
+    def get_content(cls, file_id: str) -> GetFileContentResponse:
```

Comment:
We might want to call this `download` to have symmetry with your `upload` function. Granted, from the _server's_ perspective, `get_content` makes more sense, but given that we're in the context of the client, I think it's ok to have the functions be client-centric.

Reply:
I would say it's more important to be consistent in the client, because per above, from the server's perspective it's more about getting (extracting) content, whereas from this client, you're more specifically downloading.

Reply:
Oh wait sorry what I meant was that the client methods should be named `upload`/`download`, but it makes it easier to avoid mistakes when copying DTOs, the DTOs can follow the server. Also curious if @phil-scale or Will have thoughts on this.